app/controllers/listings_controller.rb

```diff
@@ -10,6 +10,16 @@ def index
   def show
     @host_details = User.find(@listing.user_id)
     @carousel = @listing.images
+    @listing_reviews = Review
+                            .where("bookings.listing_id like #{@listing.id} and reviews.user_id != #{@listing.user_id}")
+                            .joins('JOIN bookings ON reviews.booking_id = bookings.id')
+                            .joins('JOIN users ON users.id = reviews.user_id')
+                            .paginate(page: params[:page], per_page: 3)
+    @review_average = Review
+                            .where("bookings.listing_id like #{@listing.id} and reviews.user_id != #{@listing.user_id}")
+                            .joins('JOIN bookings ON reviews.booking_id = bookings.id')
+                            .joins('JOIN users ON users.id = reviews.user_id')
+                            .average(:rating)
```

Comment:
I'm getting an error from this when I try to access the show page.
![image](https://user-images.githubusercontent.com/20994464/114380169-56f53700-9bbc-11eb-8ae5-9c86ab6fbccf.png)

Instead of doing a very long `.where` and `.joins`, you can try `.pluck` to get IDs and use them with `.where`.
Also, this can be placed in a model or helper class so that the controller can look cleaner.